### PR TITLE
test(e2e,playwright): bread-crumb.tsx

### PR DESCRIPTION
**Pull request by rohitrai3**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Solves "template-challenge-components/bread-crumb" > #51705

camperbot cant code

### CODE REVIEW

**Review by Sembauke — APPROVED** [review]

File: e2e/bread-crumb.spec.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    '/learn/foundational-c-sharp-with-microsoft/write-your-first-code-using-c-sharp/trophy-write-your-first-code-using-c-sharp'
+  );
+});
+
+test.afterEach(async ({ page }) => {
+  await page.close();
+});
+
+test.describe('Challenge Breadcrumb Component Tests', () => {
+  test('breadcrumb nav links should be visible', async ({ page }) => {
```
Comment (Ahmadkashif) [inline e2e/bread-crumb.spec.ts:14]:
```suggestion
  test('breadcrumb nav links should be visible', async ({ page }) => {
    const leftBreadcrumb = page.getByTestId('breadcrumb-left');
    const rightBreadcrumb = page.getByTestId('breadcrumb-right');
```

File: e2e/bread-crumb.spec.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    const leftBreadcrumb = page.getByTestId('breadcrumb-left');
```
Comment (Ahmadkashif) [inline e2e/bread-crumb.spec.ts:15]:
```suggestion
```

File: e2e/bread-crumb.spec.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    const rightBreadcrumb = page.getByTestId('breadcrumb-right');
```
Comment (Ahmadkashif) [inline e2e/bread-crumb.spec.ts:16]:
```suggestion
```

File: e2e/bread-crumb.spec.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    await expect(leftBreadcrumb).toBeVisible();
+    await expect(rightBreadcrumb).toBeVisible();
+  });
+
+  test('breadcrumb nav links should have correct name', async ({ page }) => {
+    const leftBreadcrumb = page.getByTestId('breadcrumb-left');
```
Comment (Ahmadkashif) [inline e2e/bread-crumb.spec.ts:22]:
```suggestion
```

File: e2e/bread-crumb.spec.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    const rightBreadcrumb = page.getByTestId('breadcrumb-right');
```
Comment (Ahmadkashif) [inline e2e/bread-crumb.spec.ts:23]:
```suggestion
```

File: e2e/bread-crumb.spec.ts [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    await expect(leftBreadcrumb).toHaveText(
+      '(New) Foundational C# with Microsoft'
+    );
+    await expect(rightBreadcrumb).toHaveText('Write Your First Code Using C#');
+  });
+
+  test('left breadcrumb nav link should have correct URL', async ({ page }) => {
+    await page.getByTestId('breadcrumb-left').click();
```
Comment (Ahmadkashif) [inline e2e/bread-crumb.spec.ts:31]: This will work, but we could improve this as well by seeing if we can identify the role as clickable list item.

**Review by Ahmadkashif — CHANGES_REQUESTED** [review]
Hi @rohitrai3 👋 

Thank you for the effort. This is some solid work 💪 

However, I have some minor suggestions that can make this PR golden.

File: e2e/bread-crumb.spec.ts (hunk 1 above)
Comment (rohitrai3) [reply e2e/bread-crumb.spec.ts:14]: Hello Ahmad, are you suggesting to remove `const` and user `page` directly in `expect`?

**Comment by rohitrai3** [comment]
@Ahmadkashif I have applied your comments, please check if they are resolved or not.

**Review by Ahmadkashif — APPROVED** [review]
For some reason, my comments were half uploaded in the last review. 

All I suggested was to move the repetitive query of both left and right breadcrumbs in the main describe block, so that we don't do that in isolated tests.

However, the current implementation looks good. 

LGTM!

**Comment by naomi-lgbt** [comment]
Congrats on your first pull request (PR)! 🎉

Thank you for your contribution to the page! 👍
We are happy to accept these changes and look forward to future contributions. 📝

**Pull request merged**